holopy/propagation/convolution_propagation.py

```diff
@@ -60,10 +57,14 @@ def propagate(data, d, index=None, wavelen=None, gradient_filter=False):
         # Propagating no distance has no effect
         return data
 
-    data = interpret_args(data, index, wavelen)
-
     if data.positions is None:
         raise MissingParameter("image spacing")
+    if wavelen is None:
+        wavelen = data.illum_wavelen
+    if index is None:
+        index = data.med_index
+    med_mavelen = wavelen/index
+
```

Comment:
This looks like another opportunity to use optical_parameters. We can then use get_med_wavelen. Additionally, the MissingParameter check (below) should be updated for med_index and illum_wavelen.